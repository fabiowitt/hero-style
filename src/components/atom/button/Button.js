import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './Button.module.scss';

const Button = ({ primary, success, danger, children }) => {
  const classes = classNames({
    [css['c-btn']]: true,
    [css['c-btn--primary']]: primary,
    [css['c-btn--danger']]: danger,
    [css['c-btn--success']]: success
  });
  return <button className={classes}>{children}</button>;
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  danger: PropTypes.bool,
  primary: PropTypes.bool,
  success: PropTypes.bool
};
Button.defaultProps = {
  danger: false,
  primary: false,
  success: false
};

export default Button;
