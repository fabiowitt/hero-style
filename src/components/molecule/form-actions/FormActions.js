import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import css from './FormActions.module.scss';

const FormActions = ({ children, alignBetween, alignEnd }) => {
  const classes = classNames({
    [css['c-form-actions']]: true,
    [css['c-form-actions--align-between']]: alignBetween,
    [css['c-form-actions--align-end']]: alignEnd
  });
  return <div className={classes}>{children}</div>;
};

FormActions.propTypes = {
  alignBetween: PropTypes.bool,
  alignEnd: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};
FormActions.defaultProps = {
  alignBetween: false,
  alignEnd: false,
  children: null
};

export default FormActions;
