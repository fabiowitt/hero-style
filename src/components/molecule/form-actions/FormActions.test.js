import React from 'react';
import { shallow } from 'enzyme';
import { action } from '@storybook/addon-actions';
import toJson from 'enzyme-to-json';

import FormActions from './FormActions';
import Button from '../../atom/button/Button';

describe('FormActions', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <FormActions>
        <Button type="primary" submit>
          Submit
        </Button>
        <Button onClick={action('cancel')}>Cancel</Button>
      </FormActions>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <FormActions alignBetween>
        <Button type="primary" submit>
          Submit
        </Button>
        <Button onClick={action('cancel')}>Cancel</Button>
      </FormActions>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match with the snapshot', () => {
    const wrapper = shallow(
      <FormActions alignEnd>
        <Button type="primary" submit>
          Submit
        </Button>
        <Button onClick={action('cancel')}>Cancel</Button>
      </FormActions>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
