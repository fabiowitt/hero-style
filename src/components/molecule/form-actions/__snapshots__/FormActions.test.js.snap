// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormActions should match with the snapshot 1`] = `
<div
  className="c-form-actions"
>
  <Button
    danger={false}
    primary={false}
    success={false}
    type="submit"
  >
    Submit
  </Button>
  <Button
    danger={false}
    onClick={[Function]}
    primary={false}
    success={false}
  >
    Cancel
  </Button>
</div>
`;

exports[`FormActions should match with the snapshot 2`] = `
<div
  className="c-form-actions c-form-actions--align-between"
>
  <Button
    danger={false}
    primary={false}
    success={false}
    type="submit"
  >
    Submit
  </Button>
  <Button
    danger={false}
    onClick={[Function]}
    primary={false}
    success={false}
  >
    Cancel
  </Button>
</div>
`;

exports[`FormActions should match with the snapshot 3`] = `
<div
  className="c-form-actions c-form-actions--align-end"
>
  <Button
    danger={false}
    primary={false}
    success={false}
    type="submit"
  >
    Submit
  </Button>
  <Button
    danger={false}
    onClick={[Function]}
    primary={false}
    success={false}
  >
    Cancel
  </Button>
</div>
`;
